utility/mm2_crawler

```diff
@@ -804,15 +804,19 @@ def try_per_file(d, hoststate, url):
     if d.files is None:
         return None
     exists = None
-    for filedetail in d.fileDetails:
-        filename = filedetail.filename
+    # d.files is a dict which contains the last (maybe 10) files
+    # of the current directory. umdl copies the pickled dict
+    # into the database. It is either a dict or nothing.
+    if not isinstance(d.files, dict):
+        return None
+    for filename in d.files.keys():
```

Comment:
You can drop the ``.keys()``, browsing a dict is much faster without it actually